lib/hmmlearn/kl_divergence.py

```diff
@@ -0,0 +1,147 @@
+"""
+
+All implementations are based upon the following:
+    http://www.fil.ion.ucl.ac.uk/~wpenny/publications/densities.ps
+"""
+
+import numpy as np
+
+from scipy.special import gammaln, digamma
+from sklearn.utils import check_array
+
+
+def kl_dirichlet(q, p):
+    """
+    KL Divergence between two dirichlet distributions
+
+    KL(q || p) = ln [gamma(q)/gamma(p)] - sum [ ln [gamma(q_j)/gamma(p_j)]
+                 - (q_j - p_j) (digamma(q_j) - digamma(p_j)]
+
+    """
+    q = np.asarray(q)
+    p = np.asarray(p)
+
+    qsum = q.sum()
+    psum = p.sum()
+    delta = q - p
+    return gammaln(qsum) - gammaln(psum) - \
+            np.sum(gammaln(q) - gammaln(p)) + \
+            np.sum(
+                np.dot(
+                    delta,
+                    (digamma(q) - digamma(qsum))
+                )
+            )
+
+
+def kl_normal_distribution(mean_q, variance_q, mean_p, variance_p):
+    """
+    KL Divergence between two normal distributions
+
+    """
+    result = (np.log(variance_p / variance_q)) / 2 \
+             + (mean_q**2 + mean_p**2 + variance_q - 2 * mean_q * mean_p) \
```

Comment:
perhaps `(mean_q - mean_p) ** 2 + (the rest)` where it's more clear that the result should be positive.